sqlmesh/core/state_sync/engine_adapter.py

```diff
@@ -1312,6 +1333,27 @@ def _intervals_to_df(
     )
 
 
+def _snapshot_interval_to_df(
+    snapshot_intervals: SnapshotIntervals,
+    is_removed: bool = False,
+) -> pd.DataFrame:
+    return pd.DataFrame(
+        [
+            _interval_to_df(
+                snapshot_intervals,
+                start_ts,
+                end_ts,
+                is_dev=is_dev,
+                is_removed=is_removed,
+            )
+            for is_dev in (False, True)
+            for start_ts, end_ts in getattr(
```

Comment:
What happens if it's empty?

Reply:
If both are empty then it would return an empty pandas dataframe. The contract here is to always return a dataframe so that seems like expected behavior. The place where this is currently used checks for empty intervals and just does nothing in that case.